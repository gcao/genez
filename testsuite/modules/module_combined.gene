#!/usr/bin/env gene run

# Test combined namespace and import usage

# Define a namespace
(ns mylib
  (var VERSION "1.0.0")
  
  (fn greet [name]
    (print "Hello, " name))
    
  (class Config
    [debug false]
    [verbose false]))

# Import from other modules (commented for now - no std library yet)
# (import "std/io")
# (import "std/math" => math)
# (import "std/string" [concat split trim])

# Another namespace
(ns myapp
  # (import "../mylib" => lib)
  
  (fn main []
    (print "Module system test completed")))

# Call the print directly for now
(print "Combined module features parsed successfully")