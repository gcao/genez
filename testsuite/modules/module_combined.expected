Combined module features parsed successfully